### PR TITLE
More cost estimation

### PR DESCRIPTION
**Pull request by justinkaseman**
### Description
- [x] Fixes request fulfillment `baseFee` not showing correctly
- [x] Cost prompt util for all write commands
- [x] Show request transaction's costs in initial estimation
- [x] Spinner, tables, & pretty output

### CODE REVIEW

**Comment by Coollaitar** [comment]
Can anybody explain to me what basically we have to do in this issue ` Cost prompt util for all write commands`?

**Comment by Coollaitar** [comment]
> Can anybody explain to me what basically we have to do in this issue ` Cost prompt util for all write commands`?

I will try my best to help you with this issue

File: tasks/Functions-client/request.js [hunk 1]
```diff
@@ -90,28 +91,36 @@ task("functions-request", "Initiates a request from an Functions client contract
     const linkBalance = subInfo[0]
     if (subInfo[0].lt(estimatedCostJuels)) {
       throw Error(
-        `Subscription ${subscriptionId} does not have sufficient funds. The estimated cost is ${estimatedCostJuels} Juels LINK, but has balance of ${linkBalance}`
+        `Subscription ${subscriptionId} does not have sufficient funds. The estimated cost is ${estimatedCostJuels} Juels LINK, but the subscription only has a balance of ${linkBalance} Juels LINK`
```
Comment (bolekk) [inline tasks/Functions-client/request.js:94]: We need to divide it by 10^18 and show a decimal LINK, otherwise the numbers are too big to eyeball

**Review by bolekk — CHANGES_REQUESTED** [review]
The utils are great!

File: tasks/Functions-client/request.js [hunk 2]
```diff
@@ -86,32 +87,41 @@ task("functions-request", "Initiates a request from an Functions client contract
...
     )
+
+    await utils.promptTxCost(transactionEstimateGas, hre, true)
+
+    // Print the estimated cost of the request
     // Ask for confirmation before initiating the request on-chain
-    const rl = readline.createInterface({
-      input: process.stdin,
-      output: process.stdout,
+    await utils.prompt(
```
Comment (KuphJr) [inline tasks/Functions-client/request.js:114]: Users will receive 2 prompts in order to initiate a request.  Can we combine these so they only have to confirm once?

**Pull request merged**